lib/ui/painting.dart

```diff
@@ -4737,24 +4737,9 @@ base class Vertices extends NativeFieldWrapperClass1 {
     List<Color>? colors,
     List<Offset>? textureCoordinates,
     List<int>? indices,
-  }) {
-    if (colors != null && colors.length != positions.length) {
-      throw ArgumentError('"positions" and "colors" lengths must match.');
-    }
-    if (textureCoordinates != null && textureCoordinates.length != positions.length) {
-      throw ArgumentError('"positions" and "textureCoordinates" lengths must match.');
-    }
-    if (indices != null) {
-      for (int index = 0; index < indices.length; index += 1) {
-        if (indices[index] >= positions.length) {
-          throw ArgumentError(
-            '"indices" values must be valid indices in the positions list '
-            '(i.e. numbers in the range 0..${positions.length - 1}), '
-            'but indices[$index] is ${indices[index]}, which is too big.',
-          );
-        }
-      }
-    }
+  }) : assert(textureCoordinates == null || textureCoordinates.length == positions.length,'"positions" and "textureCoordinates" lengths must match.'),
+    assert(colors == null || colors.length == positions.length,'"positions" and "colors" lengths must match.'),
+    assert(indices == null || indices.every((int i) => i >= 0 && i < positions.length),'"indices" values must be valid indices in the positions list.') {
```

Comment:
We could definitely let the indices check be an assert. I think at most you'll get graphical corruption.

Reply:
actually, on iOS/Metal we have to unroll triangle fans. If we end up using the indices to look up positions an invalid index could still lead to a crash.

Reply:
@jonahwilliams Do you think I should go ahead and just close this ? - or would moving the checks to the c++ code closer where they are used make more sense ?  (and possibly be faster?)